src/client/Microsoft.Identity.Client/Internal/Broker/BrokerSilentRequest.cs

```diff
@@ -88,12 +96,12 @@ internal void ValidateResponseFromBroker(MsalTokenResponse msalTokenResponse)
             {
                 _authenticationRequestParameters.RequestContext.Logger.Info(
                     LogMessages.ErrorReturnedInBrokerResponse(msalTokenResponse.Error));
-                throw new MsalServiceException(msalTokenResponse.Error, MsalErrorMessage.BrokerResponseError + msalTokenResponse.ErrorDescription);
+                throw new MsalUiRequiredException(msalTokenResponse.Error, MsalErrorMessage.BrokerResponseError + msalTokenResponse.ErrorDescription);
```

Comment:
why are these MsalUiRequiredExceptions? you should still throw serviceException right? can the developer do anything to mitigate these types of errors?

Reply:
@trwalke : if the response from the broker is an error that can be solved by the user re-signing-in (or re consenting), then it's fine to throw an `MsalUiRequiredException`
**otherwise**, if this is an error because MSAL does something wrong we should **not** throw an `MsalUiRequiredException` because this might cause an infinite loop.

Reply:
@jmprieur Depends on what the error message is. the broker may return a error if either its cache is not populated with a token or if the service returns an error. However, I just realized that I should perform a check on the account manager before sending the request to the broker to check if we have an account in the broker cache. once I do that, only service exceptions will be caught here. ill flip these back

Reply:
Are these going to be service exceptions again? If not are these causing a potential regression for the iOS broker which has already shipped?

---
In reply to: [382314137](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1649#discussion_r382314137) [](ancestors = 382314137)

Reply:
These will be flipped back. also, ios does not support silent broker auth so we are good here